src/content/developers/docs/networking-layer/index.md

```diff
@@ -0,0 +1,156 @@
+
+Discovery starts wih a game of PING-PONG. A successful PING-PONG "bonds" the new node to a boot node. The initial message that alerts a boot node to the existence of a new node entering the network is a `PING`. This `PING` includes hashed information about the new node, the boot node and an expiry time-stamp. The boot node receives the PING and returns a `PONG` containing the `PING` hash. If the `PING` and `PONG` hashes match then the connection between the new node and boot node is verified and they are said to have "bonded".
+
+Once bonded, the new node can send a `FIND-NEIGHBOURS` request to the boot node. The data returned by the boot node includes a list of peers that the new node can connect to. If the nodes are not bonded, the `FIND-NEIGHBOURS` request will fail, so the new node will not be able to enter the network.
+
+Once the new node receives a list of neighbours from the boot node, it begins a PING-PONG exchange with each of them. Successful PING-PONGs bond the new node with its neighbours, enabling message exchange.
+
+```
+start client --> connect to boot node --> bond to boot node --> find neighbours --> bond to neighbours
+```
+
+#### ENR: Ethereum Node Records {#enr}
+
+The [Ethereum Node Record (ENR)](/developers/docs/networking-layer/network addresses/) is an object that contains three basic elements: a signature (hash of record contents made according to some agreed identity scheme), a sequence number that tracks changes to the record, and an arbitrary list of key:value pairs. This is a future-proof format that allows easier exchange of identifying information between new peers and is the preferred [network address](/developers/docs/networking-layer/network-addresses) format for Ethereum nodes.
+
+#### Why is discovery built on UDP? {#why-udp}
+
+UDP does not support any error checking, resending of failed packets, or dynamically opening and closing connections - instead it just fires a continuous stream of information at a target, regardless of whether it is successfully received. This minimal functionality also translates into minimal overhead, making this kind of connection very fast. For discovery, where a node simply wants to make its presence known in order to then establish a formal connection with a peer, UDP is sufficient. However, for the rest of the networking stack, UDP is not fit for purpose. The informational exchange between nodes is quite complex and therefore needs a more fully featured protocol that can support resending, error checking etc. The additional overhead associated with TCP is worth the additional functionality. Therefore, the majority of the P2P stack operates over TCP.
+
+## DevP2P {#devp2p}
```

Comment:
Should this be a subheading of Execution stack?

```suggestion
### DevP2P {#devp2p}
```

Reply:
yes, updated